services/dataService.js

```diff
@@ -0,0 +1,67 @@
+const productMap = null;
+const categoryMap = null;
+const combinedProductMap = null;
+
+/*
+ * This method loops through the products array and maps them to a variable called productMap
+ */
+const getProducts = () => {
+  const products = require('../src/data/products');
```

Comment:
Don't load it always. Load it only when productMap is empty

Reply:
The getProducts() method is called only if the productMap is empty.C
Check the getCombinedProductMap() method.
```
const combineProductsWithCategories = () => {
  if (!productMap) {
    getProducts();
  }
  if (!categoryMap) {
    getCategories();
  }
  for (const product in productMap) {
    if (productMap.hasOwnProperty(product)) {
      combinedProductMap[productMap[product].id] = {
        ...productMap[product],
        category: categoryMap[productMap[product].categoryId],
      };
    }
  }
};
```

Reply:
Got it